### PR TITLE
Adapt logic for Custom Serving Runtime migration and adapt OVMS OOTB

### LINKED ISSUE #2059: [Story]: Revert logic of Custom Serving Runtime migration
### Goal

Rever the logic of Custom Serving runtime to only make them available in modelmesh

### Dependency issue

No dependencies

### Itemized goals

1. Rever the logic of Custom Serving runtime to only make them available in modelmesh
2. Revert OVMS support to just modelmesh

### Aspects continued elsewhere

_No response_

### Mocks

No mocks

### PR DESCRIPTION
**Pull request by lucferbux**
## Description

Closes #2059 #2048 

## How Has This Been Tested?

1. Deploy the new OVMS model
2. Clone other template and remove the `opendatahub.io/modelServingSupport` annotation
3. Check that OMVS will only supports multi model
4. Check that the Serving Runtime with no annotation only supports multi model

## Test Impact

Adapted tests and added unit testing coverage

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [X] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/modelServing/__tests__/utils.spec.ts [hunk 1]
```diff
@@ -226,3 +227,33 @@ describe('getTemplateEnabledForPlatform', () => {
     ).toBeTruthy();
   });
 });
+
+describe('getEnabledPlatformsFromTemplate', () => {
+  it('sholud return only MULTI if no annotation', () => {
```
Comment (DaoDaoNoCode) [inline frontend/src/pages/modelServing/__tests__/utils.spec.ts:232]: typo `sholud` -> `should`. Same below

Comment (lucferbux) [reply frontend/src/pages/modelServing/__tests__/utils.spec.ts:232]: agh, n00b error, thanks!

File: frontend/src/pages/modelServing/__tests__/utils.spec.ts [hunk 2]
```diff
@@ -226,3 +227,33 @@ describe('getTemplateEnabledForPlatform', () => {
...
+
+describe('getEnabledPlatformsFromTemplate', () => {
+  it('should return only MULTI if no annotation', () => {
+    const teamplateAllPlatforms = mockServingRuntimeTemplateK8sResource({
+      platforms: [],
+    });
+    expect(getEnabledPlatformsFromTemplate(teamplateAllPlatforms)).toEqual([
+      ServingRuntimePlatform.MULTI,
+    ]);
+  });
```
Comment (christianvogt) [inline frontend/src/pages/modelServing/__tests__/utils.spec.ts:239]: Could you add a test case for when the `opendatahub.io/modelServingSupport` annotation is missing. Copy this test case and delete the annotation after creating the mock serving runtime template.

Comment (lucferbux) [reply frontend/src/pages/modelServing/__tests__/utils.spec.ts:239]: oh right, forgot to add that, thank s a lot @christianvogt

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Comment by bdattoma** [comment]
when is this one going into incubation? @lucferbux

**Pull request merged**